src/platform/forms-system/test/js/review/ReviewCollapsibleChapter.unit.spec.jsx

```diff
@@ -814,4 +822,246 @@ describe('<ReviewCollapsibleChapter>', () => {
       tree.unmount();
     });
   });
+
+  describe('rendering custom content', () => {
+    const getProps = () => {
+      const CustomPage = () => <div data-testid="custom-page" />;
+      const CustomPageReview = () => <div data-testid="custom-page-review" />;
+      const pageConfig = {
+        title: '',
+        pageKey: 'test',
+        CustomPage,
+        CustomPageReview,
+        schema: { type: 'object', properties: {} },
+        uiSchema: {},
+      };
+      const pages = [
+        {
+          ...pageConfig,
+          pageKey: 'test',
+        },
+      ];
+      const chapterKey = 'test';
+      const chapter = {};
+      const form = {
+        pages: {
+          test: {
+            ...pageConfig,
+            editMode: false,
+          },
+        },
+        data: {},
+      };
+
+      return { pages, chapterKey, chapter, form };
+    };
+
+    it('should render CustomPageReview', () => {
+      const { pages, chapterKey, chapter, form } = getProps();
+      const { queryByTestId } = render(
+        <ReviewCollapsibleChapter
+          viewedPages={new Set()}
+          expandedPages={pages}
+          chapterKey={chapterKey}
+          chapterFormConfig={chapter}
+          form={form}
+          open
+        />,
+      );
+
+      expect(queryByTestId('custom-page-review')).to.exist;
+      expect(queryByTestId('custom-page')).not.to.exist;
+    });
+
+    it('should render CustomPage in edit mode', () => {
+      const { pages, chapterKey, chapter, form } = getProps();
+      form.pages.test.editMode = true;
+      const { queryByTestId } = render(
+        <ReviewCollapsibleChapter
+          viewedPages={new Set()}
+          expandedPages={pages}
+          chapterKey={chapterKey}
+          chapterFormConfig={chapter}
+          form={form}
+          open
+        />,
+      );
+
+      expect(queryByTestId('custom-page')).to.exist;
+      expect(queryByTestId('custom-page-review')).not.to.exist;
+    });
+
+    it('should render a CustomPageReview for each item in an array when showPagePerItem is true', () => {
+      const { pages, chapterKey, chapter, form } = getProps();
+      pages[0].index = 0;
+      pages.push({
+        title: '',
+        pageKey: 'test',
+        CustomPage: pages[0].CustomPage,
+        CustomPageReview: pages[0].CustomPageReview,
+        index: 1,
```

Comment:
It seems strange to me that we have an array of objects, where one of the keys in each object is `index`. But I think that's a larger forms library problem.

Reply:
Yeah, I agree. It has to do with the way the forms library deconstructs the `formConfig` and shunts different pieces of it around in different forms. :man_shrugging: Apparently, somewhere along the way, the `index` gets added to this. You can see where `page.index` is used in `ReviewCollapsibleChapter` for confirmation that it's a Thing™. I didn't make it up. :man_shrugging: